docs/docs/module_guides/workflow/index.md

```diff
@@ -0,0 +1,203 @@
+# Workflows
+
+A `Workflow` in LlamaIndex is an event-driven abstraction used to chain together several events. Workflows are made up of `steps`, with each step responsible for handling certain event types and emitting new events.
+
+`Workflow`s in LlamaIndex work by decorating function with a `@step()` decorator. This is used to infer the input and output types of each workflow for validation, and ensures each step only runs when an accepted event is ready.
+
+You can create a `Workflow` to do anything! Build an agent, a RAG flow, an extraction flow, or anything else you want.
+
+Workflows are also automatically instrumented, so you get observability into each step using tools like [Arize Pheonix](../observability/index.md#arize-phoenix-local). (**NOTE:** Observability works for integrations that take advantage of the newer instrumentation system. Usage may vary.)
+
+## Getting Started
+
+As an illustrative example, let's consider a naive workflow where a joke is generated and then critiqued.
```

Comment:
Not 100% sure if we should prioritize the class-based or free-function based approach. For now, this version feels more fleshed out I think?

Reply:
hm my pref is let's do class for now:
- each step is tied to a workflow anyways
- aesthetically it looks more organized
- it is slightly more verbose but not much 
- you can define init variables